fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -187,3 +210,41 @@ def _regression_weight_function(self, a_val, trade_off, p0, p1, a0_val):
             return trade_off / p0
         else:
             return (1 - trade_off) / p1
+
+    def _make_vector(self, formless, formless_name):
+        formed_vector = None
+        if isinstance(formless, pd.DataFrame):
+            if len(formless.columns) == 1:
+                formed_vector = formless[0].to_numpy()
+            else:
+                raise RuntimeError("%s is a DataFrame with more than one column" % formless_name)
```

Comment:
More exception types for comment

Reply:
That raises a good point, which is that if you like having the extra work of maintaining your own exception library, you can of course extend these ones to be more specific.

Reply:
I wouldn't eagerly sign up for extra work. This was just highlighting more cases where I was wondering if I was throwing the right sort of exception.